File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/bluetooth/LockPositionMapper.java
package com.lattis.ellipse.bluetooth;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.network.base.AbstractDataMapper;
import com.lattis.ellipse.domain.model.Lock;

import javax.inject.Inject;

import io.lattis.ellipse.sdk.Ellipse;

public class LockPositionMapper extends AbstractDataMapper<Lock.Hardware.Position,Ellipse.Hardware.Position> {

    @Inject
    public LockPositionMapper() {
    }

    @NonNull
    @Override
    public Ellipse.Hardware.Position mapIn(@NonNull Lock.Hardware.Position position) {
        switch (position){
            case LOCKED:
                return Ellipse.Hardware.Position.LOCKED;
            case UNLOCKED:
                return Ellipse.Hardware.Position.UNLOCKED;
            case BETWEEN_LOCKED_AND_UNLOCKED:
                return Ellipse.Hardware.Position.BETWEEN_LOCKED_UNLOCKED;
            case INVALID:
                return Ellipse.Hardware.Position.INVALID;
            default:return Ellipse.Hardware.Position.INVALID;
        }
    }

    @NonNull
    @Override
    public Lock.Hardware.Position mapOut(@NonNull Ellipse.Hardware.Position position) {
        switch (position){
            case UNLOCKED:
                return Lock.Hardware.Position.UNLOCKED;
            case LOCKED:
                return Lock.Hardware.Position.LOCKED;
            case BETWEEN_LOCKED_UNLOCKED:
                return Lock.Hardware.Position.BETWEEN_LOCKED_AND_UNLOCKED;
            case INVALID:
                return Lock.Hardware.Position.INVALID;
            default:return Lock.Hardware.Position.INVALID;
        }
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/mapper/bluetooth/LockPositionMapper.kt
package com.lattis.data.mapper.bluetooth

import com.lattis.data.mapper.AbstractDataMapper
import com.lattis.domain.models.Lock
import io.lattis.ellipse.sdk.Ellipse
import javax.inject.Inject

class LockPositionMapper @Inject constructor() :
    AbstractDataMapper<Lock.Hardware.Position?, Ellipse.Hardware.Position?>() {
    override fun mapIn(position: Lock.Hardware.Position?): Ellipse.Hardware.Position {
        return when (position) {
            Lock.Hardware.Position.LOCKED -> Ellipse.Hardware.Position.LOCKED
            Lock.Hardware.Position.UNLOCKED -> Ellipse.Hardware.Position.UNLOCKED
            Lock.Hardware.Position.BETWEEN_LOCKED_AND_UNLOCKED -> Ellipse.Hardware.Position.BETWEEN_LOCKED_UNLOCKED
            Lock.Hardware.Position.INVALID -> Ellipse.Hardware.Position.INVALID
            else -> Ellipse.Hardware.Position.INVALID
        }
    }

    override fun mapOut(position: Ellipse.Hardware.Position?): Lock.Hardware.Position {
        return when (position) {
            Ellipse.Hardware.Position.UNLOCKED -> Lock.Hardware.Position.UNLOCKED
            Ellipse.Hardware.Position.LOCKED -> Lock.Hardware.Position.LOCKED
            Ellipse.Hardware.Position.BETWEEN_LOCKED_UNLOCKED -> Lock.Hardware.Position.BETWEEN_LOCKED_AND_UNLOCKED
            Ellipse.Hardware.Position.INVALID -> Lock.Hardware.Position.INVALID
            else -> Lock.Hardware.Position.INVALID
        }
    }
}